flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobCancellationHeaders.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages;
+
+import org.apache.flink.runtime.rest.HttpMethodWrapper;
+import org.apache.flink.runtime.rest.handler.legacy.JobCancellationHandler;
+
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * Message headers for the {@link JobCancellationHandler}.
+ */
+public class JobCancellationHeaders implements MessageHeaders<EmptyRequestBody, EmptyResponseBody, JobMessageParameters> {
+
+	private static final JobCancellationHeaders INSTANCE = new JobCancellationHeaders();
+
+	private JobCancellationHeaders() {}
+
+	@Override
+	public Class<EmptyRequestBody> getRequestClass() {
+		return EmptyRequestBody.class;
+	}
+
+	@Override
+	public Class<EmptyResponseBody> getResponseClass() {
+		return EmptyResponseBody.class;
+	}
+
+	@Override
+	public HttpResponseStatus getResponseStatusCode() {
+		return HttpResponseStatus.ACCEPTED;
+	}
+
+	@Override
+	public JobMessageParameters getUnresolvedMessageParameters() {
+		return new JobMessageParameters();
+	}
+
+	@Override
+	public HttpMethodWrapper getHttpMethod() {
+		return HttpMethodWrapper.DELETE;
```

Comment:
I still think this should not be a DELETE since we can still access information about that job under the URL that we just issues the delete call to, even after the delete has finished.

Reply:
Hmm, I see your point, but I don't have good feelings to put everything behind a patch method with an endless list of query parameters. This seems somewhat wrong to me. Maybe the problem is rather that we don't distinguish between running and archived jobs on the API level. Many things don't make sense to be called on an archived job, e.g. taking a savepoint. Thus, it might make more sense to distinguish that via `/running_jobs/` and `/archived_jobs/`.

Reply:
what "endless list of query parameters" are you referring to?

Reply:
For example: `rescale`, `restart`, `pause`. Others to come in the future. Agreed that if this list contains two more methods, it wouldn't change much.